src/routes/Chat.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState, useRef } from 'react';
+import ChatForm from './ChatForm';
+import { FBrealtime } from '../fbase';
+import { onValue, ref } from '@firebase/database';
+import styled from 'styled-components';
+
+type ChatProps = {
+  userObj: any;
+  friend: any;
+};
+
+const Chat = ({ userObj, friend }: ChatProps) => {
+  const [chatObjData, setChatObjData] = useState<any>();
+  const [chatArrayData, setChatArrayData] = useState<any>([]);
+  const chatRoomID: string[] = [userObj.uid, friend.uid].sort();
+
+  //Scroll Event
+  const scrollRef: any = useRef<HTMLDivElement>();
+  useEffect(() => {
+    scrollRef.current.scrollTo(0, scrollRef.current.scrollHeight);
+  });
+
+  //DB에서 Chat data를 받아 chatObjData에 저장
+  const chatRoomRef = ref(FBrealtime, `${chatRoomID}/`);
+  useEffect(() => {
+    onValue(chatRoomRef, (snapshot) => {
+      setChatObjData(snapshot.val());
+    });
+  }, []);
+
+  //객체 형태로 chatObjData에 저장된 데이터를 배열로 바꿔주기 위한 useEffect
+  useEffect(() => {
+    const temp: object[] = [];
+    const pushChatsInArray = () => {
+      for (const key in chatObjData) {
+        temp.push(chatObjData[key]);
+      }
+      setChatArrayData(temp);
+    };
```

Comment:
clang을 주로 쓰셨어서 이렇게 명령형으로 코드가 나온 것 같아요! 
js 객체를 mapping하고싶으면 Object.keys()를 사용하면 좋을 것 같습니다! 이렇게 하면 좀 더 선언적인 코드가 될 것 같아요!
```suggestion
const pushChatsInArray = () => {
    const temp = Object.keys(chatObjData).map((key: string, idx: number) => { 
    // Object.keys는 항상 key 를 string 값으로 가지고 옵니다.
        return chatObj[key] 
        // key 가 원래 number였다면 return chatObj[Number(key)]
    })
    
    setChatArrayData(temp)
}
```

Reply:
코드만 보시구 제가 주로 쓰던 언어가 뭔지 알아채신 게 너무 신기하네요...
js식으로 객체를 매핑하는 건 느낌이 확 다르네요!!! 
훨씬 깔끔한 코드가 되네요 감사합니다!